src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -742,5 +748,103 @@ private void UpdateCacheExpirationTime(KeyValueWatcher changeWatcher, bool succe
 
             changeWatcher.CacheExpires = DateTimeOffset.UtcNow.Add(cacheExpirationTime);
         }
+
+        private async Task ExecuteWithFailOverPolicyAsync(Func<ConfigurationClient, Task> funcToExecute, CancellationToken cancellationToken = default)
+        {
+            using IEnumerator<ConfigurationClient> clientEnumerator = _configClientManager.GetAvailableClients().GetEnumerator();
+
+            // Guaranteed to return at least 1 element
+            Debug.Assert(clientEnumerator.MoveNext());
+
+            while (true)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                try
+                {
+                    await funcToExecute(clientEnumerator.Current).ConfigureAwait(false);
+
+                    _configClientManager.UpdateClientStatus(clientEnumerator.Current, successful: true);
+
+                    break;
+                }
+                catch (AggregateException ae)
+                {
+                    if (!TryFailOver(ae, clientEnumerator))
+                    {
+                        throw;
+                    }
+                }
+                catch (RequestFailedException e)
+                {
+                    if (!TryFailOver(e, clientEnumerator))
+                    {
+                        throw;
+                    }
+                }
+            }
+        }
+
+        private bool TryFailOver(Exception e, IEnumerator<ConfigurationClient> clientEnumerator)

```

Comment:
This function doesn't "try" (do) anything but a check for whether we should fail over. Suggest to change the function name to `ShouldFailOver`. I know we use the function name already. Let's rename the others. Their names kinda confusing too.

Reply:
After a second thought, maybe `CanFailOver` is a better name. You can merge what's in `ShouldFailoverForException` to this function.

Reply:
I like the suggestion `CanFailOver`, but I am leaning towards keeping the `ShouldFailOverForException` separate than this.

Reply:
The `CanFailOver` is basically a renaming of your `ShouldFailOverForException`.

Reply:
I took your other suggestion to rename it to `IsFailoverable()`. I think that makes more sense here.